src/recording/macromanager.cpp

```diff
@@ -0,0 +1,55 @@
+#include "macromanager.h"
+
+// TODO(xerus) handle track eject while recording
+// TODO(xerus) write tests
+// TODO(xerus) consider how to record the first jump
+
+/// The MacroManager handles the recording of Macros and the [MacroRecording] controls.
+MacroManager::MacroManager(EngineMaster* pEngine)
+        : m_pRecordedMacro(new Macro()) {
+    qCDebug(macros) << "MacroManager construct";
+
+    m_pToggleRecording = new ControlPushButton(
+            ConfigKey(MACRORECORDING_PREF_KEY, "recording_toggle"));
+    connect(m_pToggleRecording,
+            &ControlPushButton::valueChanged,
+            this,
+            &MacroManager::slotToggleRecording);
+    m_pCORecStatus = new ControlObject(
+            ConfigKey(MACRORECORDING_PREF_KEY, "recording_status"));
+
+    m_pCODeck = new ControlObject(ConfigKey(MACRORECORDING_PREF_KEY, "deck"));
+
+    connect(this,
+            &MacroManager::startMacroRecording,
+            pEngine,
+            &EngineMaster::slotStartMacroRecording);
+    connect(this,
+            &MacroManager::stopMacroRecording,
+            pEngine,
+            &EngineMaster::slotStopMacroRecording);
+}
+
+MacroManager::~MacroManager() {
+    qCDebug(macros) << "MacroManager deconstruct";
+    delete m_pCORecStatus;
+    delete m_pToggleRecording;
```

Comment:
Use either smart pointers or plain member variables. Manual `delete` is a hassle to maintain and easily prone to memory leaks.

Reply:
Okay, I basically copied this from another part of the code

Reply:
I guess I'll be using a shared_ptr for the Macro then and plain member variables for COs

Reply:
Use parented_ptr if it's part of the Qt object tree.